File: frontend/src/api/scramble.ts
// Server-generated scramble string. Mirrors backend/app/schemas/scramble.py.
// Public GET, no auth — rate-limited server-side (services/ratelimit.py).
//
// `scramble_token` is a signed, one-time token (HMAC over the scramble + a
// nonce + expiry) that must be threaded back on POST /solves so the server
// can lazily persist the scramble row and link solve.scramble_id to it. It
// carries no meaning client-side — just forward it opaquely.

import { request } from "./client";

export interface ScrambleOut {
  scramble: string;
  event: string;
  scramble_token: string;
}

export function fetchScramble(signal?: AbortSignal): Promise<ScrambleOut> {
  return request<ScrambleOut>("/scramble", { signal });
}

File: backend/app/schemas/scramble.py
from pydantic import BaseModel


class ScrambleOut(BaseModel):
    scramble: str
    event: str = "333"
    # HMAC-signed, one-time-use token (see app.services.scramble_token). The
    # client threads this back into POST /solves to authenticate the scramble
    # text server-side and link solve.scramble_id.
    scramble_token: str
